src/GitHub.Api/NewTaskSystem/ProcessTask.cs

```diff
@@ -134,14 +173,37 @@ public void Run()
             }
 
             if (Process.StartInfo.RedirectStandardOutput)
-                Process.BeginOutputReadLine();
+            {
+                outputReset.Reset();
+                Process.StandardOutput.BaseStream.BeginRead(
```

Comment:
Use a loop here to read the bytes and encode the string with the correct encoding. After that you can read the lines and call the appropriate output method. This way we won't need the complicated `AsyncCallback` that has a reference to itself.